integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonAnalysisRules.scala

```diff
@@ -73,3 +78,136 @@ object CarbonPreInsertionCasts extends Rule[LogicalPlan] {
+          addedTupleId = true
+          targetTable
+      }
+    } else {
+      selectPlan
+    }
+    val finalPlan = if (filter.length > 0) {
+      val alias = table.alias.getOrElse("")
+      var transformed: Boolean = false
+      // Create a dummy projection to include filter conditions
+      var newPlan: LogicalPlan = null
+      if (table.tableIdentifier.database.isDefined) {
+        newPlan = new SparkSqlParser(sparkSession.sessionState.conf).parsePlan("select * from  " +
+                                                                     table.tableIdentifier.database
+                                                                       .getOrElse("") + "." +
+                                                                     table.tableIdentifier.table +
+                                                                     " " + alias + " " +
+                                                                     filter)
+      }
+      else {
+        newPlan = new SparkSqlParser(sparkSession.sessionState.conf).parsePlan("select * from  " +
+                                                                     table.tableIdentifier.table +
+                                                                     " " + alias + " " +
+                                                                     filter)
+      }
+      newPlan transform {
+        case UnresolvedRelation(t, Some(a)) if (
+          !transformed && t == table.tableIdentifier && a == alias) =>
+          transformed = true
+          // Add the filter condition of update statement  on destination table
+          SubqueryAlias(alias, updatedSelectPlan, Option(table.tableIdentifier))
+      }
+    } else {
+      updatedSelectPlan
+    }
+    val tid = CarbonTableIdentifierImplicit.toTableIdentifier(Seq(table.tableIdentifier.toString()))
+    val tidSeq = Seq(getDB.getDatabaseName(tid.database, sparkSession))
+    val destinationTable = UnresolvedRelation(table.tableIdentifier, table.alias)
+    ProjectForUpdate(destinationTable, columns, Seq(finalPlan))
+  }
+
+  def processDeleteRecordsQuery(selectStmt: String, table: UnresolvedRelation): LogicalPlan = {
+   // val tid = CarbonTableIdentifierImplicit.toTableIdentifier(Seq(table.tableIdentifier.toString()))
+   val tidSeq = Seq(getDB.getDatabaseName(table.tableIdentifier.database, sparkSession),
+     table.tableIdentifier.table)
+    var addedTupleId = false
+    val selectPlan = new SparkSqlParser(sparkSession.sessionState.conf).parsePlan(selectStmt) transform {
+      case relation: UnresolvedRelation if (table.tableIdentifier == relation.tableIdentifier &&
+                                            addedTupleId == false) =>
+        addedTupleId = true
+        val tupleId = UnresolvedAlias(Alias(UnresolvedFunction("getTupleId",
+          Seq.empty, isDistinct = false), "tupleId")())
+        val projList = Seq(
+          UnresolvedAlias(UnresolvedStar(Option(table.alias.toSeq))), tupleId)
```

Comment:
This won't work if there is no alias to table. if there no alias then it should be passed as none